clients/venice-admin-tool/src/main/java/com/linkedin/venice/datarecovery/DataRecoveryExecutor.java

```diff
@@ -0,0 +1,104 @@
+package com.linkedin.venice.datarecovery;
+
+import static java.lang.Thread.*;
+
+import java.io.Console;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+/**
+ * DataRecoveryExecutor is the engine to run tasks in data recovery.
+ */
+public class DataRecoveryExecutor {
+  private final Logger LOGGER = LogManager.getLogger(DataRecoveryExecutor.class);
+  private final static int DEFAULT_POOL_SIZE = 10;
+  private final static int DEFAULT_POOL_TIMEOUT_IN_SECONDS = 30;
+  private final int poolSize;
+  private final ExecutorService pool;
+  private List<DataRecoveryTask> tasks;
+
+  public DataRecoveryExecutor() {
+    this(DEFAULT_POOL_SIZE);
+  }
+
+  public DataRecoveryExecutor(int poolSize) {
+    this.poolSize = poolSize;
+    this.pool = Executors.newFixedThreadPool(this.poolSize);
+  }
+
+  public void perform(Set<String> storeNames, StoreRepushCommand.Params params) {
+    String pass = getUserCredentials();
+    if (pass == null) {
+      LOGGER.error("Cannot get password, exit");
+      return;
+    }
+    params.setPassword(pass);
+
+    tasks = buildTasks(storeNames, params);
+    List<CompletableFuture<Void>> taskFutures = tasks.stream()
+        .map(dataRecoveryTask -> CompletableFuture.runAsync(dataRecoveryTask, pool))
+        .collect(Collectors.toList());
+    taskFutures.stream().map(CompletableFuture::join).collect(Collectors.toList());
+    displayTaskResult();
+
+    shutdownAndAwaitTermination();
+  }
+
+  private void shutdownAndAwaitTermination() {
+    pool.shutdownNow();
```

Comment:
So this should be `pool.shutdown()` based on the latest conversation, right?

Reply:
Thanks, it's my bad. I thought I have changed to `pool.shutdown()`...